remotes/docker/fetcher.go

```diff
@@ -210,5 +214,30 @@ func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string,
 		}
 	}
 
-	return resp.Body, nil
+	body := resp.Body
+	encoding := strings.Split(resp.Header.Get("Content-Encoding"), ", ")
+	for i := len(encoding) - 1; i >= 0; i-- {
```

Comment:
Looks like these should be treated case-insensitive, and whitespace to be trimmed; if the list has multiple items, then empty values in the list should be ignored (although that seems like a really corner-case).

https://www.rfc-editor.org/rfc/rfc9110.html#section-8.4.1

> All content codings are case-insensitive and ought to be registered within the
> "HTTP Content Coding Registry", as described in Section 16.6

Whitespace is not part of the value (so should be trimmerd); https://www.rfc-editor.org/rfc/rfc9110.html#name-field-values

> A field value does not include leading or trailing whitespace. When a specific
> version of HTTP allows such whitespace to appear in a message, a field parsing
> implementation MUST exclude such whitespace prior to evaluating the field value.

Empty values inside the list should be ignored; https://www.rfc-editor.org/rfc/rfc9110.html#section-5.6.1.2-6

> Empty elements do not contribute to the count of elements present. A recipient
> MUST parse and ignore a reasonable number of empty list elements: enough to
> handle common mistakes by senders that merge values, but not so much that they
> could be used as a denial-of-service mechanism.

Could use `strings.FieldFunc()` to split it; https://go.dev/play/p/Z8ed_yjGQS-

```go
package main

import (
	"fmt"
	"strings"
)

func main() {
	values := strings.FieldsFunc("    hello,   world,and,more,words ", func(r rune) bool {
		return r == ' ' || r == ','
	})

	fmt.Println(strings.Join(values, "|"))
}
```